File: frontend/src/services/dashboardService.js
import api from './api';

/** Role-based dashboard data. Owner: Member 4. */
const dashboardService = {
  admin: () => api.get('/dashboard/admin').then((r) => r.data),
  teacher: () => api.get('/dashboard/teacher').then((r) => r.data),
  student: () => api.get('/dashboard/student').then((r) => r.data),
};

export default dashboardService;

File: backend/src/main/java/com/group8/sams/service/DashboardService.java
package com.group8.sams.service;

import com.group8.sams.dto.response.dashboard.AdminDashboardResponse;
import com.group8.sams.dto.response.dashboard.StudentDashboardResponse;
import com.group8.sams.dto.response.dashboard.TeacherDashboardResponse;
import com.group8.sams.security.UserPrincipal;

public interface DashboardService {

    AdminDashboardResponse admin(UserPrincipal caller);

    TeacherDashboardResponse teacher(UserPrincipal caller);

    StudentDashboardResponse student(UserPrincipal caller);
}
